File: fabric/src/main/java/io/github/techtastic/vs_addon_template/fabric/VSAddonTemplateModFabric.java
package io.github.techtastic.vs_addon_template.fabric;

import net.fabricmc.api.ClientModInitializer;
import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;
import net.fabricmc.api.ModInitializer;
import org.valkyrienskies.mod.fabric.common.ValkyrienSkiesModFabric;

import static io.github.techtastic.vs_addon_template.VSAddonTemplateMod.init;
import static io.github.techtastic.vs_addon_template.VSAddonTemplateMod.initClient;

public class VSAddonTemplateModFabric implements ModInitializer {
    @Override
    public void onInitialize() {
        // force VS2 to load before eureka
        new ValkyrienSkiesModFabric().onInitialize();
        init();
    }

    @Environment(EnvType.CLIENT)
    public static class Client implements ClientModInitializer {
        @Override
        public void onInitializeClient() {
            initClient();
        }
    }
}

File: fabric/src/main/kotlin/io/github/techtastic/vs_addon_template/fabric/VSAddonTemplateModFabric.kt
package io.github.techtastic.vs_addon_template.fabric

import io.github.techtastic.vs_addon_template.VSAddonTemplateMod.init
import io.github.techtastic.vs_addon_template.VSAddonTemplateMod.initClient
import net.fabricmc.api.ClientModInitializer
import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment
import net.fabricmc.api.ModInitializer
import org.valkyrienskies.mod.fabric.common.ValkyrienSkiesModFabric

object VSAddonTemplateModFabric: ModInitializer {
    override fun onInitialize() {
        // force VS2 to load before eureka
        ValkyrienSkiesModFabric().onInitialize()

        init()
    }

    @Environment(EnvType.CLIENT)
    class Client : ClientModInitializer {
        override fun onInitializeClient() {
            initClient()
        }
    }
}
